tfjs-backend-cpu/src/kernels/Draw.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {Draw, DrawAttrs, DrawInputs, KernelConfig, KernelFunc, TypedArray} from '@tensorflow/tfjs-core';
+import {TensorInfo} from '@tensorflow/tfjs-core';
+
+import {MathBackendCPU} from '../backend_cpu';
+
+export function draw(
+    args: {inputs: DrawInputs, backend: MathBackendCPU, attrs: DrawAttrs}):
+    TensorInfo {
+  const {inputs, backend, attrs} = args;
+  const {image} = inputs;
+  const {canvas, canvasOptions, drawOptions} = attrs;
+  const alpha = drawOptions ?.alpha || 1;
+
+  const contextType = canvasOptions ?.contextType || '2d';
+  if (contextType !== '2d') {
+    throw new Error(`Context type ${
+        canvasOptions.contextType} is not supported by the CPU backend.`);
+  }
+  const ctx = canvas.getContext(contextType,
+    canvasOptions?.contextAttributes || {}) as CanvasRenderingContext2D ;
+
+  const [height, width] = image.shape.slice(0, 2);
+  const depth = image.shape.length === 2 ? 1 : image.shape[2];
+  const data = backend.data.get(image.dataId).values as TypedArray;
+  const multiplier = image.dtype === 'float32' ? 255 : 1;
+  const bytes = new Uint8ClampedArray(width * height * 4);
+
+  for (let i = 0; i < height * width; ++i) {
+    const rgba = [0, 0, 0, 255 * alpha];
+
+    for (let d = 0; d < depth; d++) {
+      const value = data[i * depth + d];
+
+      if (image.dtype === 'float32') {
+        if (value < 0 || value > 1) {
```

Comment:
Should we throw an error or just clamp it to the supported range? For webgl/webgpu backend, it's not possible to do this check and throw errors.

Reply:
For WebGL and WebGPU, I don't think we need to check values and we could just clamp them. For CPU, it's very convenient to make such checks and throwing an error would be helpful for users, because it's common that users pass values out of 255 and get a pure white canvas with no hit message. So, I think this throw-error feature is good to have.